protocol/messenger_communities.go

```diff
@@ -3637,3 +3676,42 @@ func (m *Messenger) CheckPermissionsToJoinCommunity(request *requests.CheckPermi
 
 	return m.communitiesManager.CheckPermissionToJoin(request.CommunityID, addresses)
 }
+
+func chunkSlice[T comparable](slice []T, chunkSize int) [][]T {
+	var chunks [][]T
+	for i := 0; i < len(slice); i += chunkSize {
+		end := i + chunkSize
+
+		// necessary check to avoid slicing beyond
+		// slice capacity
+		if end > len(slice) {
+			end = len(slice)
+		}
+
+		chunks = append(chunks, slice[i:end])
+	}
+
+	return chunks
+}
+
+func chunkAttachmentsByByteSize(slice []*protobuf.DiscordMessageAttachment, maxFileSizeBytes uint64) [][]*protobuf.DiscordMessageAttachment {
```

Comment:
This function doesn't seem to be used anymore.

Is this not an issue with discord file attachments that are big?
One of the reasons this chunking was introduced was because we would run into scenarios where we'd process many messages that have discord message attachments, which would take a long time to save.

Reply:
It is still used when importing the community in `RequestImportDiscordCommunity` which is a bit different than importing messages from archives.